nni/retiarii/execution/cgo_engine.py

```diff
@@ -79,12 +81,14 @@ def register_graph_listener(self, listener: AbstractGraphListener) -> None:
         self._listeners.append(listener)
 
     def _send_trial_callback(self, paramater: dict) -> None:
-        for listener in self._listeners:
-            listener.on_resource_used(0)  # FIXME: find the real resource id
+        if self.resources <= 0:
+            _logger.warning('There is no available resource, but trial is submitted.')
+        self.resources -= 1
```

Comment:
`self.resources` is supposed to manage resource, in dedup case, one trial still consumes one gpu?

Reply:
Fixed. One trial may consume multiple GPUs. It is fixed to consume how many GPUs a trial uses.